cylc/flow/remote.py

```diff
@@ -298,7 +299,10 @@ def construct_ssh_cmd(
         'CYLC_CONF_PATH',
         'CYLC_COVERAGE',
         'CLIENT_COMMS_METH',
-        'CYLC_ENV_NAME'
+        'CYLC_ENV_NAME',
+        *(glbl_cfg().get(['scheduler'])
+            ['run hosts']
+            ['ssh forward environment variables']),
```

Comment:
This configuration will apply to all SSH commands made by Cylc, not just the one made by `cylc play` and not just SSH'es too or from the scheduler `run hosts`.

For context, here are some examples of SSH use in Cylc:

* play (client => scheduler-host): Automatic distribution of workflows onto scheduler hosts.
* clean (client => remote-platform): Removal of files on remote platforms.
* job-submission (scheduler-host => remote-platform): Submit jobs to remote platforms.

Suggest moving the configuration into the `[platforms]` section:

```ini
[platforms]
  [[myplatform]]
    ssh forward environment variables = FOO, BAR, PROJECT
```

It can then be used here like so:

```suggestion
        *platform['ssh forward environment variables'],
```

Ping @hjoliver from his earlier comment. In order to configure this in `run hosts` and have it apply only to run host comms we would need to compare the FQDN of the host name we are contacting to determine whether it is in `run hosts` in the first place. The `[platforms][localhost]` section is used for all run-host SSH's where it is used to configure the `ssh command` etc. So we might as well configure this in platforms opening this functionality up to other uses right?

Reply:
Does it make sense to have different platforms with different forwarded variables? Any variable used by a specific platform will also need to be sent to the scheduler for it to work properly, I can see things becoming confusing if they are out of sync.

Reply:
I don't have any use cases in mind for per-platform configuration. It could potentially make sense, e.g. for your use case if the project codes differ from one platform to another. There might potentially be other use cases for this sort of functionality e.g. configuring things at the Cylc level which you might otherwise have to configure in shell profile files.

The options for implementation are either a per-platform configuration, or a global configuration (as implemented). I think putting the global configuration in the `run hosts` section is a bit too misleading as it also configures SSH commands which are neither to or from the run hosts.